src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -283,6 +292,19 @@ grpc_error* SecurityHandshaker::CheckPeerLocked() {
   return GRPC_ERROR_NONE;
 }
 
+void SecurityHandshaker::ScheduleRead(void* arg, grpc_error* /* error */) {
+  SecurityHandshaker* h = static_cast<SecurityHandshaker*>(arg);
+  grpc_endpoint_read(h->args_->endpoint, h->args_->read_buffer,
```

Comment:
Note that in this case, we are accessing the endpoint, read buffer, and closure without holding the lock.  I think this is fine, which is why I suggested doing the same thing in http_connect_handshaker.cc.  But if for some reason I'm wrong about that, then we probably need to do the same thing here.

Reply:
Removed it from http_connect_handshaker. I don't see an issue